utils.groovy

```diff
@@ -431,9 +431,19 @@ def AWSBuildUploadCredentialExists() {
     return utils.credentialsExist(creds)
 }
 
+def getSlackDefaultChannel() {
+    return shwrapCapture("""
+        oc get secret -n ${env.PROJECT_NAME} -o json slack-api-token | \
+          jq -r '.metadata.annotations["jenkins.io/default-channel"]'
+    """)
+}
+
 // Emits Slack message if set up, otherwise does nothing.
 def trySlackSend(params) {
```

Comment:
Personally, I think the Slack integration is a good example of how Jenkins pipelines can grow in an ad-hoc/unbounded way.  IMO, a better model is for things like chat notifications to *poll* (with opt-in event triggers) to send a message as part of a separate process (container).  This is how e.g. the Prow job alerts work as far as I know.